gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -875,6 +875,15 @@ transcribe_expression (Parser<MacroInvocLexer> &parser)
   if (expr == nullptr)
     return AST::Fragment::create_error ();
 
+  // FIXME: make this an error for some edititons
+  if (parser.peek_current_token ()->get_id () == SEMICOLON)
+    {
+      rust_warning_at (
+	parser.peek_current_token ()->get_locus (), 0,
+	"trailing semicolon in macro used in expression context");
+      parser.skip_token ();
+    }
+
```

Comment:
The FIXME at line 1050 in this file can be removed.

Reply:
But this will have to be changed to an error in future editions

Reply:
We'll keep the existing code, but emit an error or a warning depending on which `-frust-edition` option has been passed. Both behaviors need to stay. So I agree that the comment can be removed, or it can be kept for now since we'll need to touch that part of the code again